header.go

```diff
@@ -57,29 +58,18 @@ func NewHeader(cards []Card, htype HDUType, bitpix int, axes []int) *Header {
 	}
 
 	if _, ok := keys["NAXIS"]; !ok {
-		dcards = append(dcards, Card{
+		dcards = append(cards, Card{
 			Name:    "NAXIS",
 			Value:   len(hdr.Axes()),
 			Comment: "number of data axes",
 		})
-	}
-
-	if len(hdr.Axes()) >= 1 {
-		if _, ok := keys["NAXIS1"]; !ok {
-			dcards = append(dcards, Card{
-				Name:    "NAXIS1",
-				Value:   hdr.Axes()[0],
-				Comment: "length of data axis 1",
-			})
-		}
-	}
-
-	if len(hdr.Axes()) >= 2 {
-		if _, ok := keys["NAXIS2"]; !ok {
-			dcards = append(dcards, Card{
-				Name:    "NAXIS2",
-				Value:   hdr.Axes()[1],
-				Comment: "length of data axis 2",
+		nax := len(hdr.Axes())
+		for i := 0; i < nax; i++ {
+			axis := strconv.Itoa(i + 1) // index from 0, hdu starts at NAXIS1
+			dcards = append(cards, Card{
```

Comment:
```suggestion
			dcards = append(dcards, Card{
```

also, before adding that data card, you should test whether that `NAXISxxx` key already exist (and `continue` if it does).